packages/core/content-type-builder/admin/src/components/FormModalNavigationProvider/tests/index.test.js

```diff
@@ -0,0 +1,48 @@
+import { renderHook, act } from '@testing-library/react-hooks';
+import { INITIAL_STATE_DATA } from '../constants';
+import FormModalNavigationProvider from '../index';
+import useFormModalNavigation from '../../../hooks/useFormModalNavigation';
+
+const removeFunctionsFromObject = state => {
+  const stringified = JSON.stringify(state);
+  const parsed = JSON.parse(stringified);
+
+  return parsed;
+};
+
+describe('FromModalNavigationProvider', () => {
+  it('sets the initial state', () => {
+    const { result } = renderHook(() => useFormModalNavigation(), {
+      wrapper: FormModalNavigationProvider,
+    });
+
+    const currentStateWithoutFunctions = removeFunctionsFromObject(result.current);
+
+    expect(currentStateWithoutFunctions).toEqual(INITIAL_STATE_DATA);
+  });
+
+  it('updates the form navigation state when selecting a custom field', () => {
+    const { result } = renderHook(() => useFormModalNavigation(), {
+      wrapper: FormModalNavigationProvider,
+    });
+
+    act(() => {
+      result.current.onClickSelectCustomField({
```

Comment:
@soupette is this what you had in mind for tests? Sorry not confident I did this correctly 😬 thanks for your help!

Reply:
You can wrap the Provider around a component that prints out the state and generate a snapshot, but this seems to be working as well.